### PR TITLE
Three-argument open used

### PR DESCRIPTION
**Pull request by IvanLudvig**
Replaced two-argument open calls in 13 files to three-argument calls. #75

### CODE REVIEW

File: bin/awk [hunk 1]
```diff
@@ -45,15 +45,15 @@ usage() unless @ARGV;
 # pipes or even redirections for process control
 # on broken soi-disant operating systems.
 
-open(SAVE_OUT, ">&STDOUT") || die "can't save stdout: $!";
+open(SAVE_OUT, '<', ">&STDOUT") || die "can't save stdout: $!";
 die unless defined fileno SAVE_OUT;
 
-open(TMPIN, ">  " . ($tmpin = "a2pin.$$"))	     ||
-open(TMPIN, ">  " . ($tmpin = "/tmp/a2pin.$$"))      ||
+open(TMPIN, '<', ">  " . ($tmpin = "a2pin.$$"))	     ||
```
Comment (briandfoy) [inline bin/awk]: These look suspect. I think you've done the wrong thing here.

File: bin/awk [hunk 2]
```diff
@@ -126,7 +126,7 @@ die "empty program" unless -s $tmpout;
 seek(TMPOUT, 0, 0)	   || die "can't rewind $tmpout: $!";
 $program = do { local $/;  <TMPOUT> };
 close(TMPOUT)	   || die "can't close $tmpout: $!";
-open(STDOUT, ">&SAVE_OUT") || die "can't restore stdout: $!";
+open(STDOUT, '<', ">&SAVE_OUT") || die "can't restore stdout: $!";
```
Comment (briandfoy) [inline bin/awk]: Check this one

File: bin/awk [hunk 3]
```diff
@@ -105,7 +105,7 @@ unshift @ARGV, @vargs;  # put back var=val statements
 
 close(TMPIN)		    || die "can't close $tmpin: $!";
 
-open(STDOUT, ">&TMPOUT")    || die "can't dup to $tmpout: $!";
+open(STDOUT, '<', ">&TMPOUT")    || die "can't dup to $tmpout: $!";
```
Comment (briandfoy) [inline bin/awk]: Check this one

File: bin/awk [hunk 4]
```diff
@@ -45,15 +45,15 @@ usage() unless @ARGV;
...
 
-open(TMPIN, ">  " . ($tmpin = "a2pin.$$"))	     ||
-open(TMPIN, ">  " . ($tmpin = "/tmp/a2pin.$$"))      ||
+open(TMPIN, '<', ">  " . ($tmpin = "a2pin.$$"))	     ||
+open(TMPIN, '<', ">  " . ($tmpin = "/tmp/a2pin.$$"))      ||
 die "can't find a tmp input file";
 
-open(TMPOUT,"+> " . ($tmpout = "a2pout.$$"))	     ||
-open(TMPOUT,"+> " . ($tmpout = "/tmp/a2pout.$$"))    ||
+open(TMPOUT,'<', "+> " . ($tmpout = "a2pout.$$"))	     ||
```
Comment (briandfoy) [inline bin/awk]: Check these

File: bin/awk [hunk 5]
```diff
@@ -45,15 +45,15 @@ usage() unless @ARGV;
...
 # on broken soi-disant operating systems.
 
-open(SAVE_OUT, ">&STDOUT") || die "can't save stdout: $!";
+open(SAVE_OUT, '<', ">&STDOUT") || die "can't save stdout: $!";
 die unless defined fileno SAVE_OUT;
 
-open(TMPIN, ">  " . ($tmpin = "a2pin.$$"))	     ||
-open(TMPIN, ">  " . ($tmpin = "/tmp/a2pin.$$"))      ||
+open(TMPIN, '<', ">  " . ($tmpin = "a2pin.$$"))	     ||
+open(TMPIN, '<', ">  " . ($tmpin = "/tmp/a2pin.$$"))      ||
```
Comment (briandfoy) [inline bin/awk]: check these

File: bin/awk [hunk 6]
```diff
@@ -45,15 +45,15 @@ usage() unless @ARGV;
 # pipes or even redirections for process control
 # on broken soi-disant operating systems.
 
-open(SAVE_OUT, ">&STDOUT") || die "can't save stdout: $!";
+open(SAVE_OUT, '<', ">&STDOUT") || die "can't save stdout: $!";
```
Comment (briandfoy) [inline bin/awk]: check these

File: bin/look [hunk 7]
```diff
@@ -69,7 +69,7 @@ $search = squish($search);
 
 FILE: for my $file (@dicts) {
 
-    unless (open(DICT, "< $file")) {
+    unless (open(DICT, '<', "< $file")) {
```
Comment (briandfoy) [inline bin/look]: check this

File: bin/par [hunk 8]
```diff
@@ -230,7 +230,7 @@ find sub {
     FILE
 
     # Inline file as here-doc.
-    open F, "< $_" or die "Couldn't open '$_': $!";
+    open F, '<', "< $_" or die "Couldn't open '$_': $!";
```
Comment (briandfoy) [inline bin/par]: check this

File: bin/split [hunk 9]
```diff
@@ -71,7 +71,7 @@ sub nextfile {
 
     # MS-DOS: $curname = "$prefix." . $curext;
     $curname = $prefix . $curext;
-    open (FH, ">$curname") or die "$me: Can't open $curname: $!\n";
+    open (FH, '<', ">$curname") or die "$me: Can't open $curname: $!\n";
```
Comment (briandfoy) [inline bin/split]: check this

File: bin/split [hunk 10]
```diff
@@ -105,7 +105,7 @@ clue if $opt{"?"};    # heeeeeeeeeeelp!
 my $infile = (defined $ARGV[0] ? $ARGV[0] : "-");
 my $prefix = ($ARGV[1] ? "$ARGV[1]" : "x");
 
-open (INFILE, "$infile") || die "$me: Can't open $infile: $!\n";
+open (INFILE, '<', "$infile") || die "$me: Can't open $infile: $!\n";
```
Comment (briandfoy) [inline bin/split]: we can also get rid of the quotes around the filename

File: bin/tee [hunk 11]
```diff
@@ -37,7 +37,7 @@ unless ($nostdout) {
 $| = 1 if $unbuffer;
 
 for (@ARGV) {
-    if (!open($fh, (/^[^>|]/ && $mode) . $_)) {
+    if (!open($fh, '<', (/^[^>|]/ && $mode) . $_)) {
```
Comment (briandfoy) [inline bin/tee]: check this

File: bin/uuencode [hunk 12]
```diff
@@ -80,7 +80,7 @@ sub encode($$) {
     printf "begin %03o $destination\n", $mode || 0644;
 
     local *INPUT;
-    open(INPUT, "< $source") 	|| die "can't open $source: $!";
+    open(INPUT, '<', "< $source") 	|| die "can't open $source: $!";
```
Comment (briandfoy) [inline bin/uuencode]: check this

File: bin/wc [hunk 13]
```diff
@@ -230,7 +230,7 @@ sub wc_fh {
 
 if (@ARGV) {
     foreach my $filename (@ARGV) {
-        open(FH, "< $filename") or die "cannot open $filename: $!\n";
+        open(FH, '<', "< $filename") or die "cannot open $filename: $!\n";
```
Comment (briandfoy) [inline bin/wc]: check this

**Comment by IvanLudvig** [comment]
I'm sorry for mixing it all up. Thanks for pointing out every mistake. I've added corrections

File: bin/tee [hunk 14]
```diff
@@ -37,7 +37,7 @@ unless ($nostdout) {
 $| = 1 if $unbuffer;
 
 for (@ARGV) {
-    if (!open($fh, (/^[^>|]/ && $mode) . $_)) {
+    if (!open($fh, '/^[^>|]/', ($mode) . $_)) {
```
Comment (briandfoy) [inline bin/tee]: This one is suspect. Maybe leave this one alone.

Not that the stuff in parens is trying to determine the mode then append that to `$_`. The mode is not the pattern `/^[^>|]/`.

File: bin/what [hunk 15]
```diff
@@ -31,7 +31,7 @@ else
 {
     foreach $file (@ARGV)
     {
-        open(FILE, "<$file") or die "Unable to read $file: $!";
+        open(FILE, '<', "<$file") or die "Unable to read $file: $!";
```
Comment (briandfoy) [inline bin/what]: This one is wrong.

File: bin/awk [hunk 16]
```diff
@@ -45,15 +45,15 @@ usage() unless @ARGV;
...
 
-open(TMPIN, ">  " . ($tmpin = "a2pin.$$"))	     ||
-open(TMPIN, ">  " . ($tmpin = "/tmp/a2pin.$$"))      ||
+open(TMPIN, '>',  ($tmpin = "a2pin.$$"))	     ||
+open(TMPIN, '>',  ($tmpin = "/tmp/a2pin.$$"))      ||
 die "can't find a tmp input file";
 
-open(TMPOUT,"+> " . ($tmpout = "a2pout.$$"))	     ||
-open(TMPOUT,"+> " . ($tmpout = "/tmp/a2pout.$$"))    ||
+open(TMPOUT, "+> ",  ($tmpout = "a2pout.$$"))	     ||
```
Comment (briandfoy) [inline bin/awk]: take the spaces out of the mode.

File: bin/awk [hunk 17]
```diff
@@ -45,15 +45,15 @@ usage() unless @ARGV;
...
+open(TMPOUT, "+> " , ($tmpout = "/tmp/a2pout.$$"))    ||
```
Comment (briandfoy) [inline bin/awk]: take the spaces out of the mode.

**Comment by IvanLudvig** [comment]
Did the changes

**Comment by briandfoy** [comment]
I'm tracking this, but I want to improve the tests to make sure everything happened correctly. That's just a limit on my time. I think the changes are good, but I also know that bugs don't care what I think. :)

**Comment by briandfoy** [comment]
I manually applied many of these changes because I needed to adjust other things to make them work out. Thanks for the contributions!

**Pull request closed without merging**